src/instructlab/train/linux_train.py

```diff
@@ -222,6 +223,8 @@ def linux_train(
     )
 
     test_dataset = load_dataset("json", data_files=os.fspath(test_file), split="train")
+    train_dataset = ensure_legacy_dataset(train_dataset)
```

Comment:
I think you need to wrap the call to ensure_legacy_dataset in a try-except block otherwise we can't gently catch the `ValueError` from `is_messages_dataset`.

Reply:
Good call, I'll add that.